test/e2e/tests/collectibles.spec.js

```diff
@@ -0,0 +1,200 @@
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .transaction-list-item:nth-of-type(1)',
+          { timeout: 10000 },
+        );
+
+        // Verify transaction
+        const completedTx = await driver.findElement('.list-item__title');
+        const completedTxText = await completedTx.getText();
+        assert.equal(completedTxText, 'Send Token');
+      },
+    );
+  });
+  it('should approve an address to transfer a single NFT', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: 'connected-state',
+        ganacheOptions,
+        smartContract,
+        title: this.test.title,
+        failOnConsoleError: false,
+      },
+      async ({ driver, _, contractRegistry }) => {
+        const contract = contractRegistry.getContractAddress(smartContract);
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // Click approve
+        await driver.openNewPage(`http://127.0.0.1:8080/?contract=${contract}`);
+        await driver.waitForSelector({
+          css: '#collectiblesStatus',
+          text: 'Deployed',
+        });
+        await driver.delay(veryLargeDelayMs);
+        await driver.fill('#approveTokenInput', '1');
+        await driver.clickElement('#approveButton');
+        await driver.waitUntilXWindowHandles(3);
+        const windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        // Verify dialog
+        const title = await driver.findElement(
+          '[data-testid="confirm-approve-title"]',
+        );
+        const data = await driver.findElements(
+          '.confirm-approve-content__data .confirm-approve-content__small-text',
+        );
+        assert.equal(
+          await title.getText(),
+          'Give permission to access your TestDappCollectibles (#1)?',
+        );
+        assert.equal(await data[0].getText(), 'Function: Approve');
+
+        // Confirm approval
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+        await driver.clickElement('[data-testid="home__activity-tab"]');
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .transaction-list-item:nth-of-type(1)',
+          { timeout: 10000 },
+        );
+
+        // Verify transaction
+        const completedTx = await driver.findElement('.list-item__title');
+        const completedTxText = await completedTx.getText();
+        assert.equal(completedTxText, 'Approve Token spend limit');
+      },
+    );
+  });
+  it('should approve an address to transfer all NFTs', async function () {
```

Comment:
Could we add a test for revoking `setApprovalForAll`?

Reply:
Yes we can, that will require changes in the [test-dapp](https://github.com/MetaMask/test-dapp/blob/main/src/index.js#L548) as it's hardcoded to true, so we will have to do that as a follow up.

Reply:
https://github.com/MetaMask/metamask-extension/issues/15488